### PR TITLE
Define extension value as registered extension name

### LINKED ISSUE #477: Referencing extensions in ocfl_layout.json
When describing `ocfl_layout.json`, the draft OCFL spec reads:

> key - A key identifying the precise arrangement of directories and OCFL objects under the storage root, i.e. how OCFL object identifiers are mapped to directory hierarchies. The value of this key is not defined in the OCFL specification, but MUST correspond to a value given in an Extension.

The text is unclear what "value" the key must correspond to. Opening the issue here rather than under extensions because my assumption, which may be incorrect, is that the value it is supposed to correspond to is the file name of the extension (eg `0002-fixed-length-n-tuple-trees.md` or `0002-fixed-length-n-tuple-trees`) and that this should be clarified in the spec.

### PR DESCRIPTION
**Pull request by zimeon**
Fixes #477 ... maybe ;-)

### CODE REVIEW

**Comment by ahankinson** [comment]
Just flagging @pwinckles here so that he can see this.

File: draft/spec/index.html [hunk 1]
```diff
@@ -975,30 +975,30 @@ <h2>Root Structure</h2>
...
         </p>
         <ul>
           <li>
-            <code>key</code> - A key identifying the precise arrangement of directories and OCFL objects under the
-            storage root, i.e. how OCFL object identifiers are mapped to directory hierarchies. The value of this key
-            is not defined in the OCFL specification, but <span id="E071">MUST</span> correspond to a value given in an
-            <a>Extension</a>.
+            <code>extension</code> - An extension name that identifies an arrangement of directories and OCFL objects
+            under the storage root, i.e. how OCFL object identifiers are mapped to directory hierarchies. The
+            value of the <code>extension</code> key <span id="E071">MUST</span> be the registered <a>extension</a>
```
Comment (awoods) [inline draft/spec/index.html:994]: There is still room for misinterpretation of what is meant by "registered extension name". I would suggest we refer to the proposed `Extension ID` seen in this pull-request: https://github.com/OCFL/extensions/pull/16/files#diff-77315547024f9923f13fb0a37b4ecf3aR3

Comment (zimeon) [reply draft/spec/index.html:994]: Or maybe we should fix that PR to say extension name? I'm loathe to call something that isn't a URI or similar an id

Comment (pwinckles) [reply draft/spec/index.html:994]: I don't mind updating that PR. I added that line speculatively.

Comment (zimeon) [reply draft/spec/index.html:994]: cf. "Digest Algorithm Name" also in https://ocfl.github.io/extensions/0001-digest-algorithms.html

Comment (awoods) [reply draft/spec/index.html:994]: Using a URI would be nice. 
It is important that the language here be very specific, unambiguous.

Comment (neilsjefferies) [reply draft/spec/index.html:994]: https://github.com/OCFL/extensions/pull/13 also has updates around extension names.

**Comment by ahankinson** [comment]
Does that mean that this can be merged?

**Comment by ahankinson** [comment]
Nevermind -- saw the comment in Slack

**Pull request merged**